gateway-js/src/datasources/types.ts

```diff
@@ -21,6 +22,11 @@ export type GraphQLDataSourceProcessOptions<
    * The request to send to the subgraph.
    */
   request: GatewayGraphQLRequestContext<TContext>['request'];
+
+  /**
+   * The path to the current processed node within the graph query
+   */
+  nodeGraphPath?: ResponsePath;
```

Comment:
I'd move this inside the `INCOMING_OPERATION` part (that path makes no sense for a `HEALTH_CHECK` or `LOADING_SCHEMA` kind of request) and maybe rename it to something like `pathInIncomingRequest` (we call `incomingRequestContext` the total operation received by the gateway, and the "path" is really the path within that gateway operation at which this particular subgraph "sub-query" fits in, and that's probably how I would describe that field btw; I also wouldn't use `node` in the name as that's query plan specific naming and most users aren't meant to be familiar with that).

Could be nice to also document that the path may be `undefined` if it is unavailable, but will be empty (rather than `undefined`) for "top-level" fetches.

Reply:
Thanks for the clarifications! I've adopted your guidelines.